src/app/SERVICE/tasks.service.ts

```diff
@@ -1,48 +1,50 @@
 import { Injectable } from '@angular/core';
 import { Task } from '../models/task.model';
-import { BehaviorSubject, Observable, map } from 'rxjs';
+import { Observable, map, switchMap } from 'rxjs';
 import { HttpClient } from '@angular/common/http';
 import { Router } from '@angular/router';
-import { Constants } from '../utils/constants';
+import { tasksApiUrl } from '../utils/constants';
 
 @Injectable({
   providedIn: 'root'
 })
 export class TasksService {
 
-  constructor(private http: HttpClient, private router: Router, private constants: Constants) {
+  constructor(private http: HttpClient, private router: Router) {
   }
 
-  public getAllTasks():Observable<Task[]>{
-    return this.http.get<Task[]>(this.constants.tasksApiUrl)
+  getAllTasks(): Observable<Task[]> {
+    return this.http.get<Task[]>(tasksApiUrl)
       .pipe(
-        map(data => { return data })
+        map(data => data)
       );
   }
 
-  async createNewTask(task: Task) {
+  createNewTask(task: Task) {
     if (task) {
-      await this.http.post(this.constants.tasksApiUrl, task).subscribe()
+      this.http.post(tasksApiUrl, task).subscribe()
     } else {
       return
     }
   }
-  async deleteTask(deleteTaskId: string) {
-
-    this.http.delete<string>(this.constants.tasksApiUrl + deleteTaskId).subscribe(() => {
-    })
+  deleteTask(taskId: string){
+    return this.http.delete<Task>(tasksApiUrl + taskId)
+    .pipe(
+      switchMap(() => this.getAllTasks()),
+    );
   }
-  async taskDetails(tasksId: string) {
-   console.log(tasksId)
+  taskDetails(tasksId: string) {
+    console.log(tasksId)
   }
-  async updateTask(tasksNewData: Task) {
+  updateTask(taskNewData: Task) {
     const body = {
-      id: tasksNewData.id,
-      description: tasksNewData.description,
-      title: tasksNewData.title,
-      status: tasksNewData.status
+      id: taskNewData.id,
+      description: taskNewData.description,
+      title: taskNewData.title,
+      status: taskNewData.status
     }
-    await this.http.patch<string>(this.constants.tasksApiUrl, body).subscribe()
+    this.http.patch<string>(tasksApiUrl, body)
```

Comment:
you would have to return this statement here as well, basically all http requests should return an observable, which you can then use in components as you see fit

Reply:
then if you need, you can subscribe to them in components and unsubscribe when you're done with them (usually you'd unsubscribe in the ngOnDestroy() method)